library/src/main/java/com/novoda/downloadmanager/lib/DefaultsDownloadManagerModules.java

```diff
@@ -0,0 +1,114 @@
+import com.novoda.downloadmanager.notifications.CompleteNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.DownloadingNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.EmptyCancelledNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.EmptyCompleteNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.EmptyFailedNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.FailedNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.NotificationImageRetriever;
+import com.novoda.downloadmanager.notifications.OkHttpNotificationImageRetriever;
+import com.novoda.downloadmanager.notifications.QueuedNotificationCustomiser;
+
+class DefaultsDownloadManagerModules implements DownloadManagerModules {
+
+    private final Context context;
+    @Nullable
+    private final QueuedNotificationCustomiser queuedCustomiser;
+    @Nullable
+    private final DownloadingNotificationCustomiser downloadingCustomiser;
+    @Nullable
+    private final CompleteNotificationCustomiser completeCustomiser;
+    @Nullable
+    private final CancelledNotificationCustomiser cancelledCustomiser;
+    @Nullable
+    private final FailedNotificationCustomiser failedCustomiser;
+    @Nullable
+    private final DownloadClientReadyChecker readyChecker;
+    @Nullable
+    private final NotificationImageRetriever imageRetriever;
+
+    DefaultsDownloadManagerModules(Context context) {
+        this(context, null, null, null, null, null, null, null);
+    }
+
+    public DefaultsDownloadManagerModules(Context context,
+                                          @Nullable QueuedNotificationCustomiser queuedCustomiser,
+                                          @Nullable DownloadingNotificationCustomiser downloadingCustomiser,
+                                          @Nullable CompleteNotificationCustomiser completeCustomiser,
+                                          @Nullable CancelledNotificationCustomiser cancelledCustomiser,
+                                          @Nullable FailedNotificationCustomiser failedCustomiser,
+                                          @Nullable DownloadClientReadyChecker readyChecker,
+                                          @Nullable NotificationImageRetriever imageRetriever) {
+        this.context = context;
+        this.queuedCustomiser = queuedCustomiser;
+        this.downloadingCustomiser = downloadingCustomiser;
+        this.completeCustomiser = completeCustomiser;
+        this.cancelledCustomiser = cancelledCustomiser;
+        this.failedCustomiser = failedCustomiser;
+        this.readyChecker = readyChecker;
+        this.imageRetriever = imageRetriever;
+    }
+
+    @Override
+    public NotificationImageRetriever getNotificationImageRetriever() {
+        if (imageRetriever == null) {
+            return new OkHttpNotificationImageRetriever();
```

Comment:
still think this may be better done in the builder - once you do "builder.build()" I feel like you're supposed to have a fully-instantiated class